client/src/components/History/Content/GenericElement.vue

```diff
@@ -1,23 +1,30 @@
 <script setup lang="ts">
-import { type PropType, ref, set } from "vue";
+import { type PropType, type Ref, ref, set } from "vue";
 
 import { type components } from "@/api/schema";
 
 import ContentItem from "./ContentItem.vue";
 
+type ElementObject = components["schemas"]["DCObject"];
+type Element = components["schemas"]["DCESummary"];
+
 defineProps({
     dsc: {
-        type: Object as PropType<components["schemas"]["DCObject"]>,
+        type: Object as PropType<ElementObject>,
         required: true,
     },
 });
 
-const expandCollections = ref({});
-const expandDatasets = ref({});
+const expandCollections: Ref<Record<string, boolean>> = ref({});
```

Comment:
```suggestion
const expandCollections = ref<Record<string, boolean>>({});
```
Maybe this is a personal preference, but somehow this reads cleaner to me and doesn't require the additional `Ref` type import.

Reply:
Of course - agreed completely!